docker/daemon/daemon_src.go

```diff
@@ -45,29 +42,14 @@ func newImageSource(ctx *types.SystemContext, ref daemonReference) (types.ImageS
 	}
 	defer inputStream.Close()
 
-	// FIXME: use SystemContext here.
-	tarCopyFile, err := ioutil.TempFile(tmpdir.TemporaryDirectoryForBigFiles(), "docker-daemon-tar")
+	src, err := tarfile.NewSource("", inputStream)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "Error getting source")
 	}
-	defer tarCopyFile.Close()
-
-	succeeded := false
-	defer func() {
-		if !succeeded {
-			os.Remove(tarCopyFile.Name())
-		}
-	}()
-
-	if _, err := io.Copy(tarCopyFile, inputStream); err != nil {
-		return nil, err
-	}
-
-	succeeded = true
 	return &daemonImageSource{
 		ref:         ref,
-		Source:      tarfile.NewSource(tarCopyFile.Name()),
-		tarCopyPath: tarCopyFile.Name(),
+		Source:      src,
+		tarCopyPath: src.GetTarPath(),
```

Comment:
`tarCopyPath` only exists for deleting the temporary file on `Close`.  Shouldn’t that functionality also be moved from `docker/daemon` to `docker/tarfile`?

Reply:
Moved to docker/tarfile